### DISCUSSION TITLE
Help for training a multi speaker model for voice cloning

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#276] [Archived]
<USER>>>> julian.weber<br/> <DATE> [September 24, 2020, 1:42pm]<br/> <br/> Hi,<br/> <br/> I'm training a model in french to do voice cloning and I'm using<br/> CorentinJ's encoder to compute the speaker embedding. I started to train<br/> on the 5 voices of the french mailabs and got good results. It's not<br/> sufficient to reproduce accent or anything but the voice generated is<br/> ineligible and the pitch matches with the cloned speaker.<br/> <br/> To increase the fidelity of the cloning I tried adding the 'mix' folder<br/> of the mailabs dataset. (Each chapter of the book is read by a different<br/> speaker) and then the quality of the voice started to go downhill<br/> quickly but I didn't noticed since everything seemed fine on the tensor<br/> board. (grey and orange curves) *slash* <br/> <br/> I then (because the tensorboard was fine) added the french part of the<br/> common voice dataset set where a speaker only has around 4 voice samples<br/> (12k different speakers). The tensorboard told me that the model<br/> struggled to make the loss as small as before but it was understandable<br/> since the data is much more noisy and every speaker has a different mic.<br/> <br/> I was a bit horrified by the sythetised results after 260k steps the<br/> models can't form one word correctly in a sentence. Enven the checkpoint<br/> before common voice. So I have a few questions:<br/> <br/> a specific speaker ? Or test sentences at all for that matter since<br/> even though I specified test_sentences_file in config.json, I<br/> coulnd't see any of my test sentences in the tensorboard. (it worked<br/> before on a single speaker model)<br/> <br/> every sample has its speaker embedding computed ? (I have my<br/> original speakers but all other non identified speaker are under the<br/> mix speaker)<br/> <br/> preprocessing that can help ? Audio normalization ?<br/> <br/> (in the config.json we can read : )<br/> <br/> > 'stats_path': null // DO NOT USE WITH MULTI_SPEAKER MODEL. scaler<br/> > stats file computed by 'compute_statistics.py'. If it is defined,<br/> > mean-std based normalization is used and other normalization params<br/> > are ignored<br/> <br/> Thanks a lot for reading this post<br/> <br/> Note: For big datasets I find the eval interval impractical, I trained<br/> for 260k steps and the last eval was around 190k. Wouldn't it be<br/> interesting to set a eval interval based on training steps and not epoch<br/> ?<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/help-for-training-a-multi-speaker-model-for-voice-cloning]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [September 24, 2020, 6:31pm]<br/> <br/> I don't think you can have test sentences with specific speakers, but I<br/> might be wrong. I do not think I have seen a switch. But you might be<br/> able to write some lines and use a delimiter in the test.txt file. I<br/> don't think it is trouble what you are saying about the mix speaker.<br/> Since every utterance has its own embedding and I doubt every training<br/> batch separates speakers. For normalization, I personally have used the<br/> default configuration. Re: the mean-var, I think Eren mentioned it is a<br/> bit of a hassle to compute mean-var stats for multispeaker, because<br/> these have to be separate for each speaker.<br/> <br/> The multispeaker TTS works ok (I have run some tests), but my opinion is<br/> that it absolutely benefits from restoring pretrained weights on it. So<br/> maybe first you'd like to train a French single speaker and then restore<br/> that. However, do bear in mind that in an ideal scenario the speech is<br/> very clean. What I advise is to start with English instead, because then<br/> you can restore the LJSpeech TTS and either use LibriTTS or VCTK.<br/> has gotten very<br/> impressive results on VCTK and he has even been able to form artificial<br/> speakers.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> julian.weber<br/> <DATE> [September 29, 2020, 12:12pm]<br/> <br/> Thanks for your answer, I wasn't able to get good cloning results from<br/> the collab notebook. Like erogol said in this<br/> [comment](https://github.com/mozilla/TTS/issues/485#issue-672584017),<br/> the clonned voice is not very close to the original speaker. But I think<br/> this will be fixed when we have a better speaker encoder.<br/> <br/> I feel like it was the attention mechanism that failed me during<br/> training, do you know of an easy way to freeze the weights of certain<br/> part of the network during training ?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [September 29, 2020, 12:29pm]<br/> <br/> once taught me<br/> he has it implemented on his fork so that you can toggle freezing<br/> through the config.json, but if you want to use the original repo, you<br/> can go in train_tts.py and, e.g. if you want to freeze different parts:<br/> <br/> for param in model.postnet.parameters():<br/> param.requires_grad = False<br/> <br/> for name, param in model.decoder.named_parameters():<br/> param.requires_grad = False<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> julian.weber<br/> <DATE> [September 29, 2020, 12:34pm]<br/> <br/> Great thank you so much. I'll keep you posted here if I get good results<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [September 29, 2020, 12:36pm]<br/> <br/> No problem! But I'd give restoring pretrained weights a chance<br/> it always helps me no matter what<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> julian.weber<br/> <DATE> [September 29, 2020, 12:43pm]<br/> <br/> You mean restoring only part of the models that you interested in ? I<br/> was going to restore my model to 70k wait for the loss to stabilize and<br/> then freeze the weights (I'm thinking attention and stopnet) and add the<br/> other datasets<br/> <br/> [Archived Post]<br/>
